R/install.snowball.R

```diff
@@ -72,69 +72,130 @@ install.snowball <- function(pkg, date, include.suggests, force.install = FALSE,
-          message1("Attempt #3 to install binary from from MRAN")
-        }
-      } # end MRAN
+      }
 
 
       # 3.7 INSTALL K FROM SOURCE IF SUPPOSED TO, OR I STILL NOT INSTALLED
       if (from.k == "source" | !is.pkg_vrs.installed(pkg.k, vrs.k)) {
-        install.source(pkg_vrs.k, lib.k, date, quiet.install = quiet.install)
-        message1("Attempting to install from source")
+        message1("\n>Attempting to install from source")
+
+        # 1) Try CRAN
+        # If current version
+        if ((pkg_vrs.k %in% .pkgenv[["current.packages"]]$pkg_vrs) == TRUE) {
+          url1 <- paste0("https://cran.r-project.org/src/contrib/", pkg_vrs.k, ".tar.gz")
+          install.packages(url1, repos = NULL, lib = lib.k, type = "source", dependencies = F, quiet = quiet.install)
+        }
+
+        # If archive version
+        if ((pkg_vrs.k %in% .pkgenv[["current.packages"]]$pkg_vrs.k) == FALSE) {
+          url2 <- paste0("https://cran.r-project.org/src/contrib/Archive/", pkg.k, "/", pkg_vrs.k, ".tar.gz")
+          install.packages(url2, repos = NULL, lib = lib.k, type = "source", dependencies = F, quiet = quiet.install)
+        }
+
+        # 2) Try MRAN
+        # Attempt 3 overall, MRAN around selected date, current tarballs path
+        if (is.pkg_vrs.installed(pkg.k, vrs.k) == FALSE) # if still not installed
+          {
+            # Set date for mran, midpoint of available dates around requested date  +-10 days,
+            date <- as.Date(date) # Ensure formated as date
+            max.date <- min(Sys.Date() - 2, date + 10) # use as highest possible mran date, 2 days ago
+            mran.date <- get.available.mran.date(date - 10, max.date) # Function get.available.mran.date() in utils.R
+
+            # Try path for current packages
+            url3 <- paste0("https://cran.microsoft.com/snapshot/", mran.date, "/src/contrib/", pkg_vrs.k, ".tar.gz")
+            install.packages(url3, repos = NULL, lib = lib.k, type = "source", dependencies = F, quiet = quiet.install)
+          }
+        # Attempt 4 overall, MRAN around selected date, Archive tarballs path
+        if (is.pkg_vrs.installed(pkg.k, vrs.k) == FALSE) # if still not installed
+          {
+            url4 <- paste0("https://cran.microsoft.com/snapshot/", mran.date, "/src/contrib/Archive/", pkg.k, "/", pkg_vrs.k, ".tar.gz")
+            install.packages(url4, repos = NULL, lib = lib.k, type = "source", dependencies = F, quiet = quiet.install)
+          }
+
+        # Attempt 5, MRAN around Publish date, current path
+        if (is.pkg_vrs.installed(pkg.k, vrs.k) == FALSE) # if still not installed
+          {
+            # Set date for mran, midpoint of available dates around requested date  +-10 days,
+            toc.pkg <- toc(pkg)
+            date.Published <- as.Date(subset(toc.pkg, Version = vrs)$Published) + 2
```

Comment:
```suggestion
            date.Published <- toc.pkg$Published[toc.pkg$Version == vrs] + 2
```

The `Published` column is supposed to already be a `Date`